src/osp/UserInputHandler.cpp

```diff
@@ -315,40 +213,41 @@ ButtonControlHandle UserInputHandler::config_get(std::string const& name)
             //             | (uint8_t(varCfg.m_invert) << 1)
             //             | (uint8_t(varCfg.m_nextOp) << 2);
 
-            control.m_exprPress.emplace_back(btnInsert.first, varCfg);
+            rControl.m_exprPress.emplace_back(varCfg.create(btnInsert.first));
         }
 
         // New control has been created, now return a pointer to it
 
-        return ButtonControlHandle(this, m_controls.size() - 1);
+        return EButtonControlIndex(m_btnControls.size() - 1);
     }
 }
 
-MouseMovementHandle UserInputHandler::mouse_get()
+void UserInputHandler::button_unsubscribe(EButtonControlIndex index)
 {
-    return MouseMovementHandle(this);
-}
+    uint16_t &rRefCount = m_btnControls.at(size_t(index)).m_referenceCount;
```

Comment:
i feel like this should instead be handled with an std::shared_ptr ?

Reply:
I guess it's possible, but the elements being reference counted are in a vector. I also don't want there to be a reference counted object for every single control, hence using integer indices and ControlSubscriber instead of the Handle classes.